examples/algorithm_progression.py

```diff
@@ -0,0 +1,172 @@
+import argparse
+import concurrent.futures
+import inspect
+import itertools
+import os
+from pathlib import Path
+
+import algorithms
+import numpy as np
+import yaml
+
+import large_image
+
+VARIABLE_LAYERS = ['z', 'c', 't']
+
+
+argparser = argparse.ArgumentParser(
+    prog='Algorithm Progression',
+    description='Apply an algorithm to an input image '
+                'for every parameter set in a given parameter space',
+)
+argparser.add_argument(
+    'algorithm_code',
+    choices=algorithms.ALGORITHM_CODES.keys(),
+    help='Code to specify which of the available algorithms should be used',
+)
+argparser.add_argument('input_filename', help='Path to an image to use as input')
+argparser.add_argument(
+    '-o',
+    '--output_dir',
+    required=False,
+    help='Name for a new directory in this location, wherein result images will be stored',
+)
+argparser.add_argument(
+    '-w',
+    '--num_workers',
+    required=False,
+    default=4,
+    type=int,
+    help='Number of workers to use for processing alogrithm iterations',
+)
+argparser.add_argument(
+    '-p',
+    '--param',
+    action='append',
+    required=False,
+    nargs='*',
+    help='A parameter to pass to the algorithm; '
+         'instead of using the default value, '
+         'Pass every item in the number space, '
+         'specified as `--param=param_name,start,end,num_items[,open]`',
+)
+
+
+def apply_algorithm(algorithm, input_filename, output_dir, params, unique_params):
+    source = large_image.open(input_filename)
+    filename = f'{algorithm.__name__}_{"_".join(["{:.2f}".format(v) for v in unique_params])}.tiff'
```

Comment:
`.2f` isn't guaranteed to be enough precision (e.g., --param=hue_value,0,1,256,open` will have collisions.

Reply:
Changed in [8f12e60](https://github.com/girder/large_image/pull/1051/commits/8f12e60fa621de791b9463118c999c13ddb2e88a)